client/src/utils/authFetch.js

```diff
@@ -0,0 +1,15 @@
+export async function getProtectedData(path) {
+  try {
+    const response = await fetch(path, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    if (!response.ok) throw new Error(response.status);
```

Comment:
This checks if the response is not the 2xx range, but the calling code may want to handle errors in the 4xx range rather than silencing them (like display an snackbar if the match link is invalid or expired, etc)